rfc/plugins.md

```diff
@@ -0,0 +1,206 @@
+# Plugins for ember-cli-deploy
+
+This document describes the API related to plugins in the next
+version of ember-cli-deploy.
+
+### Overview of plugins
+
+A plugin is an ember-cli addon that hooks into the ember-cli-deploy
+pipeline in order to add functionality to a deploy. Example addon
+functionality would be uploading assets to S3, writing the index.html to
+Redis, or notifying a Slack channel a deploy has completed.
+
+In general, OSS plugins should focus on doing a specific task. Most
+Ember developers with common deployment targets will compose multiple
+plugins to fine tune their deployment process. Developers with very
+custom needs might create a single private plugin that implements all
+aspects of their deployment process within the structure provided by
+ember-cli-deploy.
+
+Because plugins are implemented via addons, they may be included via
+node_module dependencies or via in-repo-addons.
+
+### Identifying plugins
+
+Plugins are ember-cli addons. They will also have the keyword
+`ember-cli-deploy-plugin`. An example `package.json` for a plugin:
+
+```
+{
+  "name": "ember-cli-deploy-example-plugin",
+  "version": "0.4.0",
+  // ...
+  "devDependencies": {
+    "ember-cli": "0.2.0",
+    "ember-cli-deploy": "0.4.1"
+    // ...
+  },
+  "keywords": [
+    "ember-addon",
+    "ember-cli-deploy-plugin"
```

Comment:
Something to think about here, and I don't believe it's mentioned anywhere else in this document, but this keyword isn't really relevant other than to tell indicate to people (or machines) looking at `package.json` that this is an ember-cli-deploy addon.

Our code actually won't see this keyword as, by the time we get access to the addons via `this.project.addons`, we don't have access to the package.json (this is my understanding anyway).

It is for this reason that we have previously set a `type` property on the plugin object and the `adapter-registry` interrogates this property ([here](https://github.com/ember-cli/ember-cli-deploy/blob/master/lib/utilities/adapter-registry.js#L36)) to see whether the addon is an ember-cli-deploy plugin.

Maybe we should mention this in the document.